presto-hive/src/main/java/io/prestosql/plugin/hive/parquet/ParquetPageSource.java

```diff
@@ -91,10 +91,12 @@ public Page getNextPage()
 
             Block[] blocks = new Block[fields.size()];
             for (int fieldId = 0; fieldId < blocks.length; fieldId++) {
-                Type type = types.get(fieldId);
-                blocks[fieldId] = fields.get(fieldId)
-                        .map(field -> (Block) new LazyBlock(batchSize, new ParquetBlockLoader(field)))
-                        .orElse(RunLengthEncodedBlock.create(type, null, batchSize));
```

Comment:
Why not to use `Optional.orElseGet()`? Your change would be less invasive.

Reply:
That would seem to be worse for performance (since it's a capturing lambda) and the conditional check here seems straightforward as-is. If people feel strongly, I can make that change.

Reply:
Then please add a comment to prevent refactoring in opposite direction.

How did you find this? Did it cause any issues?

I see your point. Since creating a lambda is a cost. Maybe that cost is comparable to creating this run length encoded block (there is almost no memory allocation there). That would mean there is no need to actually change this code at all. WDYT?

Reply:
I don't have comparable benchmarks for PrestoSQL before and after this change, only PrestoDB ([link](https://github.com/prestodb/presto/pull/13834)) and limited to a single experiment run on a pair of EMR clusters (where performance between clusters can vary significantly, so take this with a big pile of salt). In that unscientific experiment, queries that were mostly straight-shot reads over the whole file were consistently about 50% faster.

Now, that code was doing something much more wasteful so I wouldn't expect the difference to be as dramatic in PrestoSQL, but the code inside the loop runs per column, per `ParquetPageSource#getNextPage` call so doing unnecessary work can definitely hurt throughput.

Reply:
Thank you for your explanation.